10-JavaScript/js-in-the-browser.md

```diff
@@ -1,146 +1,91 @@
+- Know some ways to do basic manipulation of the DOM in JavaScript
 
-## Running JavaScript in the Developer Console
+## Browsers Can Run JS
 
-Our first step is to prove that our browser Google Chrome can read and execute JavaScript.
+By the late 80s, software engineering was already a huge industry full of research, money, and interest. However, most of the money in consumer/enterprise software was in dedicated applications/programs.
 
-In your browser, open a new empty tab, and pull up your developer tools. Click on the Console tab. This is where calls to `console.log()` will end up. It's also where you'll see any errors that occur.
+However, the early 90s brought the beginning of the web. What happens when you combine all of the following factors?
 
-Notice also that there is something that looks like a command prompt. Let's see what it does.
+- A set of emerging browser technologies around the web that caused major interest and competition between large companies
+- A lack of standards, process of standardization, or regulation around web technologies
 
-```javascript
-console.log('live, from the browser!');
-```
+Events in the 90s led to the major browser, Netscape, and thus its competitors to adopt support for JavaScript. Meaning, **since early in the history of web, browsers became environments that could read and execute JavaScript code.**
 
-Nice, there it is. OK, let's do something a little more interesting - a popup box.
+The development and evolution of browsers, browser features, and JavaScript's language and utilities have been intertwined. Browsers, by necessity, support and run JavaScript, and evolve to do so in nicer and more efficient ways.
 
-```javascript
-alert('four score and seven years ago...');
-```
+This should feel special to us because browsers do not run other languages like this "out of the box" (without special tools, libraries, or circumstances). Our Rails web apps never asked _the browser_ to run any Ruby code; it was always our servers (or our machines/computers that acted as web servers) that needed to execute Ruby. Our servers simply **gave responses with website/HTML data,** which our browsers are very good at doing.
 
-Pops right up! Can't do that in the terminal!
+JavaScript is now:
+- **The** programming language for modern web
+- Standardized and open
+- An evolved language with many more features
 
-[Where did the `alert` method come from? Who defined it?](https://developer.mozilla.org/en-US/docs/Web/API/Window/alert) `Window` is something that the browser makes available to us. When we write JavaScript that runs in the browser, we can predict that the `alert` method will be available to us.
+### Exploring Chrome Dev Tools
 
-What happens if we make a mistake? Let's reference an undefined variable.
+Let's prove to ourselves that JavaScript can run in the browser by executing JavaScript in the browser. We can do this with Chrome Dev Tools's console. It doesn't run Ruby code.
```

Comment:
It feels a little weird to say it specifically doesn't run Ruby code.  It makes it sound like it maybe supports multiple languages and Ruby just isn't one.

```suggestion
Let's prove to ourselves that JavaScript can run in the browser by executing JavaScript in the browser. We can do this with Chrome Dev Tools's console (it only runs JavaScript code).
```